assets/js/googlesitekit/widgets/index.js

```diff
@@ -19,10 +19,102 @@
 /**
  * Internal dependencies
  */
+import { dispatch, select } from 'googlesitekit-data';
+import { STORE_NAME } from './datastore/constants';
 // This import has side-effects; it registers the Widgets datastore on the default
 // data store registry (eg. `googlesitekit.data`).
 import './datastore';
 
-const Widgets = {};
+const Widgets = {
+	/**
+	 * Register a widget area
+	 *
+	 * @since n.e.x.t
+	 * @param {string}             slug               Widget Area's slug.
+	 * @param {Object}             settings           Widget Area's settings.
+	 * @param {number}             settings.priority  Priority for this widget area. Default: 10.
+	 * @param {string}             settings.title     Title for this widget area.
+	 * @param {string}             settings.subtitle  Subtitle for this widget area.
+	 * @param {(string|undefined)} settings.icon      URL to SVG icon for this widget area.
+	 * @param {string}             settings.style     Widget area style (one of "boxes", "composite").
+	 * @param {(string|Array)}     areas              Widget Area slug(s).
+	 */
+	registerWidgetArea( slug, settings, areas ) {
+		dispatch( STORE_NAME ).registerWidgetArea( slug, settings );
+		dispatch( STORE_NAME ).assignWidget( slug, areas );
+	},
+
+	/**
+	 * Register a widget.
+	 *
+	 * @since n.e.x.t
+	 * @param {string}         slug      Widget's slug.
+	 * @param {Object}         settings  Widget's settings.
+	 * @param {(string|Array)} contexts  Widget Context's slug(s).
+	 */
+	registerWidget( slug, settings, contexts ) {
```

Comment:
See above, can we name the last parameter `contextSlugs` like in the datastore?

Reply:
updated